gui/app/periodic_graph_painter.cpp

```diff
@@ -0,0 +1,77 @@
+#include "periodic_graph_painter.h"
+#include "periodic_graph.h"
+#include "qnanocolor.h"
+#include <QDebug>
+#include <QVarLengthArray>
+#include <QtMath>
+#include <algorithm>
+#include <math.h>
+PeriodicGraphPainter::PeriodicGraphPainter() {}
+
+void PeriodicGraphPainter::paint(QNanoPainter *m_painter) {
+  float w = width();
+  float h = height();
+
+  int size = dataset_.size();
+
+  if (size <= 2)
```

Comment:
2 points should technically be enough to draw a line chart, right?

Reply:
True. I was using spline before I found out that I had more points than pixels.